### ISSUE TITLE
Change running user

### ISSUE THREAD

**Issue by wouterverduin** [nextcloud/nextcloud-snap#384]
Hey guys,

Loving the snap so far however i am running into a problem: I got snap running in a VM and my data on another VM (freenas). When i change the data mountpoint to my mounted Freenas it doesnt have the right permissions. 
I figured if i changed the running user to either root or 1001 (mathcing the root of freenas) it can write to that folder.

My question is: How do i do this? Or are there other ways around?

**Comment by shaokuuipo** [comment]
You can try mount your freenas under the /media folder, and running command:
sudo snap connect nextcloud:removable-media
Now, try edit the path of the data directory to that folder under the /media

**Comment by kyrofa** [comment]
Yeah, to clarify what @shaokuuipo said: the snap's services already run as root, but the has has very confined disk access. If you connect the interface mentioned, you can access external disks mounted in /media/, but barring that the snap can essentially only access /var/snap/nextcloud/ .

**Comment by shaokuuipo** [comment]
Maybe you can create Symbolic Link   point to your data folder under /media

**Comment by kyrofa** [comment]
AppArmor is smart enough to know when symlinks are used, and I believe it will resolve them when determining access capabilities, so that probably won't work.

**Comment by wouterverduin** [comment]
Figured the symbolic link would be the quickest fix for my problem, however after i made the symbolic link it shows up in my shell only not in my webinterface of nextcloud... Is that because it isnt capable of working with it?

Is the option of mounting it at /media/ a working solution?

**Comment by kyrofa** [comment]
Yes, the option of mounting in /media is a working solution (it's what I do myself).

**Comment by wouterverduin** [comment]
Thanks for your reply. I adjusted my settings as mentioned. Mounted using 'sudo snap connect nextcloud:removable-media' and moved my data folder to /media/NextCloud. So far so good, however:

I noticed my .ocdata file disappeared after moving the files. I made a new one but as expected i got the 'data directory not writable' error. 
When i look i see the folders for Nextcloud got these permissions: drwxrwxr-x+
And the .ocdata i made: -rwxrwxr-x+

I cant chmod the file because it says 'permission denied'. The owner of the files is 1001 (account of my freenas user). 

@kyrofa How did you manage to work around this error? Or is it a mistake to use a subfolder and should i use the folder '/media'?

Thanks in advance!
